executor/include/executor/property/base_property.hpp

```diff
@@ -14,29 +14,86 @@
 
 namespace executor {
 
-template <typename Derived, bool requireable, bool preferable>
-struct basic_executor_property {
-  // Part of Proposal P1393R0
+/**
+ * \brief A base class for all properties, containing the common functionality
+ * needed by all properties
+ *
+ * \details A base class was needed as each property needed to specify a minimum
+ * set of functionality such as is_applicable_property, static query,
+ * is_requirable, is_preferable etc. which would lead to code duplication
+ * without a base class.
+ *
+ * A CRTP design is used as static query needs to be able to access the
+ * DerivedProperty in order to query the overloaded query member function in an
```

Comment:
`in order to query the overloaded query member function`

Reword this part to not have 2 meanings of query: the English verb and the member function

Reply:
`in order to call the overloaded query member`
Better?